restli-server/src/main/java/com/linkedin/restli/internal/server/model/ResourceModel.java

```diff
@@ -164,6 +198,38 @@ public ResourceModel(final Class<? extends RecordTemplate> valueClass,
          namespace);
   }
 
+  /**
+   * Constructor.
+   *
+   * @param valueClass resource value class
+   * @param resourceClass resource class
+   * @param parentResourceClass parent resource class
+   * @param name resource name
+   * @param resourceType {@link ResourceType}
+   * @param namespace namespace
+   * @param d2ServiceName The d2 service name for the resource
+   */
+  public ResourceModel(final Class<? extends RecordTemplate> valueClass,
```

Comment:
Is this new constructor ever used? No need to introduce this variant if not used.

Reply:
This new one is used by the annotation reader.